packages/manager/apps/web/client/app/hosting/database/order/public/hosting-database-order-public.service.js

```diff
@@ -0,0 +1,105 @@
+import sortBy from 'lodash/sortBy';
+import toNumber from 'lodash/toNumber';
+
+import { PRODUCT_NAME } from './hosting-database-order-public.constants';
+
+export default class {
+  /* @ngInject */
+  constructor(
+    $q,
+    $translate,
+    OvhApiHostingWeb,
+    OvhApiMe,
+    OvhApiOrder,
+    OvhApiOrderCatalogPublic,
+  ) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.OvhApiHostingWeb = OvhApiHostingWeb;
+    this.OvhApiMe = OvhApiMe;
+    this.OvhApiOrder = OvhApiOrder;
+    this.OvhApiOrderCatalogPublic = OvhApiOrderCatalogPublic;
+  }
+
+  computeCharacteristics(characteristicsOfAvailableProducts) {
+    return characteristicsOfAvailableProducts.map((characteristics) => ({
+      display: this.$translate.instant(
+        characteristics.quantity === 1
+          ? 'ovhManagerHostingDatabaseOrderPublic_customization_characteristics_display_1'
+          : 'ovhManagerHostingDatabaseOrderPublic_customization_characteristics_display_+',
+        {
+          diskSpace: characteristics.diskSpace,
+          quantity: characteristics.quantity,
+        },
+      ),
+      characteristics,
+    }));
+  }
+
+  static buildPlanCode(diskSpace, quantity) {
+    return `sql_perso_${diskSpace}_${quantity}`;
+  }
+
+  async getCharacteristicsOfAllProducts(serviceName) {
+    return this.OvhApiOrder.CartServiceOption()
+      .v6()
+      .get({
+        productName: PRODUCT_NAME,
+        serviceName,
+      })
+      .$promise.then((products) =>
+        sortBy(
+          products
+            .filter((product) => product.planCode.startsWith('sql_perso'))
+            .map((product) => {
+              const [, , diskSpace, quantity] = product.planCode.split('_');
+
+              return {
+                diskSpace: toNumber(diskSpace),
+                quantity: toNumber(quantity),
+              };
+            }),
+          ['diskSpace', 'quantity'],
+        ),
+      );
+  }
+
+  async getUnavailableDiskSpaces(serviceName) {
+    return this.OvhApiHostingWeb.ExtraSqlPerso()
+      .v6()
+      .query({
+        serviceName,
+      })
+      .$promise.then((extraSqls) =>
+        extraSqls.map((extra) => toNumber(extra.match(/\d+/)[0])),
+      );
+  }
+
+  async getCatalog() {
+    return this.OvhApiMe.v6()
+      .get()
+      .$promise.then(
+        (user) =>
+          this.OvhApiOrderCatalogPublic.v6().get({
+            productName: PRODUCT_NAME,
+            ovhSubsidiary: user.ovhSubsidiary,
+          }).$promise,
+      );
+  }
+
+  async getCharacteristicsOfAvailableProducts(serviceName) {
```

Comment:
This will imply `$timeout`ing every use, no ?

Reply:
I've removed, it's an old codebase, sometimes I don't know why things are this way